### PR TITLE
Build.sh improvement

### PR DESCRIPTION
**Pull request by jonisc**
Now you just have to execute ./build.sh versionfolder to create a new .deb. Basically just an improvement. 
Tried fixing .DS_Store issues (delete all .DS_Store files in Gandalf folder in build.sh) but I need some confirmation by @mehulrao. 

Several small improvements. Not only @1Conan 's improvements. 
Will check if folder exists and files are present. You can argue about this, but it's just better to do so in my opinion.

### CODE REVIEW

File: Gandalf/build.sh [hunk 1]
```diff
@@ -84,9 +129,13 @@ echo "Copying the DEBIAN scripts"
...
 cp "postinst" "${PKG_PACKAGE}/DEBIAN"
+echo
+read -p  "${BOLD}${RED}--- If you have anything else you need to put into Gandalf, now's the time. If you're finished or you don't want to add something press any key to continue. ---${NORMAL}"
+echo
+
+# Remove .DS_Store created by macOS (experimental)
 
-echo "If you have anything else you need to put into Gandalf, now's the time. You have 15 seconds from the moment this message appears."
-sleep 15
+find . -name '.DS_Store' -type f -delete
```
Comment (1Conan) [inline Gandalf/build.sh]: Nothing different with my `find`. You just changed `-exec rm {}`. `-exec rm {}` is actually much more recommended.

File: Gandalf/build.sh [hunk 2]
```diff
@@ -1,36 +1,81 @@
...
-FIRM=$(cat $VER/firmware.txt)
+
+# Define script styling (tput)
+
+BOLD=$(tput bold)
+NORMAL=$(tput sgr0)
+RED=$(tput setaf 1)
+
+# Check if started without parameters
+if [ "${1}" = "" ]; then
```
Comment (1Conan) [inline Gandalf/build.sh]: https://google.github.io/styleguide/shell.xml

`${1}` should be `$1`.

Also, I believe you're declaring a variable.

File: Gandalf/build.sh (hunk 1 above)
Comment (jonisc) [inline Gandalf/build.sh]: Ok. Then I'll use your find.

Comment (jonisc) [inline Gandalf/build.sh]: Question you use `find . -type f -name '.DS_STORE' -exec rm {} +`
What's the + for?

**Comment by jonisc** [comment]
@1Conan ok. So I fixed many things. But I can't put all the constants & variables to the top: Otherwise there'll be some cat errors (cat can't find file or folder -> error). If I abort the script before defining e.g. PKG_CONFLICTS there won't be this errors. It's just cosmetic… but if you have a better idea you can improve the branch.

**Comment by jonisc** [comment]
Improvements by @1Conan in #51.

**Comment by jonisc** [comment]
Ok. He wants to merge into builds branch and not into master.

**Comment by jonisc** [comment]
@mehulrao: There´s something wrong with your App: If you compile Gandalf (102/x) via eg. Ubuntu there´ll be errors: there are then 2 Folders. One called Gandalf.app with capital G, and an other called gandalf.app with a lowercase g. So Ubuntu copies the Gandalf.app and if you install it on iOS the app won´t show. I´ll compile your app via my mac, and we´ll see what´s happening...

**Comment by jonisc** [comment]
Didn't compile your app. Just extracted it from a the latest official .deb. 

There were some errors. Hopefully I squashed them now. One error I think we must discuss is this one:
[Dpkg wants to remove /var/mobile/Downloads](https://imgur.com/a/uhUqd)

**Comment by EthanRDoesMC** [comment]
That's a dpkg thing that dpkg does. If downloads were empty, it'd try to remove /mobile. Which it can't. Because there are things in it.

**Comment by EthanRDoesMC** [comment]
To build Gandalf, you need macOS or an OS that recognizes the .app extension in the same way. But since macOS is not preloaded with the stuff required to make it, there will need to be YET ANOTHER FREAKING SCRIPT (lol) to install the stuff on macOS.

**Comment by EthanRDoesMC** [comment]
I figure this is okay for tonight.

**Comment by jonisc** [comment]
I could build it on Ubuntu ;-). You just have to make sure that there's only a Gandalf.app folder.

**Comment by jonisc** [comment]
@ Everyone @EthanRDoesMC @mehulrao
Just to clarify for anybody. 
Attention: syntax has changed: now you must cd to the folder where build.sh is and execute it with ./build.sh version:

`./build.sh x` will build GandalfX, `./build.sh 102` will build Gandalf102.

**Pull request merged**